hardware/scripts/proxy.js

```diff
@@ -41,33 +50,28 @@ const HardwareProxy = (function HardwareProxy() {
 
     // Write Content
     //
-    // Our mcu firmware is a little stupid. It assumes the board size is 64x32,
-    // but our board size is 32x32.
-    // To make the embedders works easier, our HardwareProxy expect buffer sized
-    // for a 32x32 board - which means those buffers needs to be converted to
-    // the format expected by the mcu firmware.
-    // All this happens at the cost of performance. It should be fine for now but
-    // if we continue to prototype with this board and expect 60 fps it will be 
-    // definitively better to have a new firmware.
+    // Our firmware assume we have 3 16x32 chained boards handled as a unique
+    // 16x96 board.
+    // Additionally the 2nd board is rotated.
+    //
+    // The logic here is converting the coordinates from 32x48 to 16x96.
     //
-    let step = Constants.Width * Hardware.Constants.PixelSize;
-    for (let i = 0; i < inBuffer.length; i+=step) {
-      let iteration = i / step * 2;
-
-      for (let j = 0; j < step; j++) {
-        let outIndex = Hardware.Constants.HeaderSize + (iteration * step) + j;
-        outBuffer[outIndex] = 0x00;
-      }
-
-      for (let j = step; j < (step * 2); j+=Hardware.Constants.PixelSize) {
-        let outIndex = Hardware.Constants.HeaderSize + (iteration * step) + j;
-        let inIndex = i + (j - step);
-
-        outBuffer[outIndex + 0] = inBuffer[inIndex + 0];
-        outBuffer[outIndex + 1] = inBuffer[inIndex + 1];
-        outBuffer[outIndex + 2] = inBuffer[inIndex + 2];
-      }
+    // All this happens at the cost of performance. It should be fine for now
+    // but if we continue to prototype with this board and expect 60 fps it
+    // will be definitively better to have a new firmware.
+
+    // 512 is how many leds we have in 1 board.
+    const step = 512 * PixelSize;
+    outBuffer.set(inBuffer.subarray(0, step), HeaderSize);
+    const pixelsInMiddleBoard =
+      splitTypedArray(inBuffer.subarray(step, step * 2), 3);
+    let pixelsCount = 0;
+    for (const pixel of pixelsInMiddleBoard) {
+      pixelsCount += PixelSize;
+      const where = HeaderSize + step * 2 - pixelsCount;
+      outBuffer.set(pixel, where);
     }
+    outBuffer.set(inBuffer.subarray(step * 2, step * 3), HeaderSize + step * 2);
```

Comment:
boards 1 and 3 are super simple, we just copy the data from inBuffer to outBuffer (using `set` and `subarray` we should be super fast); board 2 is rotated so we set the information in outBuffer backwards using the generator created by splitTypedArray.